### PR TITLE
[Tests-Only] Refactor api scenario outline examples

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

The API acceptance tests are taking longer-and-longer as we add more coverage. In some places, we have Scenario Outline with a significant list in the Examples table. This PR refactors the current API Scenario Outlines that tests a list of edge-cases(for example, using various unusual file or folder names, using various unusual group names...) and converts it into a single scenario with a table in each step.

## Related Issue

- Part of https://github.com/owncloud/core/issues/38371

## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiProvisioning-v1/addUser.feature [hunk 1]
```diff
@@ -81,23 +95,30 @@ Feature: add user
...
+    When the administrator sends a user creation request for the following users with password using the provisioning API
+      | username        | password |
+      | Brand-New-User1 | %alt1%   |
+      | BRAND-NEW-USER2 | %alt1%   |
+      | brand-new-user3 | %alt1%   |
+      | brand-NEW-user4 | %alt1%   |
+      | BrAnD-nEw-UsEr5 | %alt1%   |
+    Then the OCS status code of responses on all endpoints should be "100"
+    And the HTTP status code of responses on all endpoints should be "200"
+    Then the following users should exist
```
Comment (Talank) [inline tests/acceptance/features/apiProvisioning-v1/addUser.feature:108]:
```suggestion
    And the following users should exist
```

**Comment by saw-jan** [comment]
LGTM

**Review by saw-jan — COMMENTED** [review]
LGTM

**Comment by kiranparajuli589** [comment]
lgtm :+1:

**Review by Talank — COMMENTED** [review]
LGTM

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 2]
```diff
@@ -3204,7 +3300,7 @@ public function groupHasBeenCreatedOnDatabaseBackend($group) {
 	 * @throws \Exception
 	 */
 	public function theseGroupsHaveBeenCreated(TableNode $table) {
-		$this->verifyTableNodeColumns($table, ['groupname']);
+		$this->verifyTableNodeColumns($table, ['groupname'], ["comment"]);
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/Provisioning.php:3303]: lets do the inverted commas the same:
```suggestion
		$this->verifyTableNodeColumns($table, ['groupname'], ['comment']);
```

File: tests/acceptance/features/apiProvisioning-v1/addUser.feature [hunk 3]
```diff
@@ -81,23 +95,30 @@ Feature: add user
...
-    And user "BrAnD-nEw-UsEr" should exist
-    And the display name of user "brand-new-user" should be "<display-name>"
-    Examples:
-      | display-name   |
-      | Brand-New-User |
-      | BRAND-NEW-USER |
-      | brand-new-user |
-      | brand-NEW-user |
-      | BrAnD-nEw-UsEr |
+  Scenario: admin creates a user with username that contains capital letters
```
Comment (individual-it) [inline tests/acceptance/features/apiProvisioning-v1/addUser.feature:98]: that is not quite the same test anymore
in the original test we create the user with a specific lower/upper case combination and expect it being available in any other lower/upper case combination. e.g. creating user `Brand-New-User` but the user `BRAND-NEW-USER` should exist.
In the changed version of the test it creates all types of users and check if they all exist.
So I think we should not refactor this test

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 4]
```diff
@@ -315,6 +315,25 @@ public function pushToLastOcsCodesArray($ocsStatusCode) {
 		\array_push($this->lastOCSStatusCodesArray, $ocsStatusCode);
 	}
 
+	/**
+	 * @return void
+	 */
+	public function pushToLastStatusCodesArray() {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/FeatureContext.php:321]:
```suggestion
	public function pushToLastStatusCodesArrays() {
```

Comment (individual-it) [reply tests/acceptance/features/bootstrap/FeatureContext.php:321]: please write also some comments about what the function does

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**